osu.Game/OsuGame.cs

```diff
@@ -313,6 +315,46 @@ private void load()
         /// <param name="beatmapId">The beatmap to show.</param>
         public void ShowBeatmap(int beatmapId) => waitForReady(() => beatmapSetOverlay, _ => beatmapSetOverlay.FetchAndShowBeatmap(beatmapId));
 
+        public void SelectBeatmap(int beatmapId)
+        {
+            var map = GetBeatmapFromId(beatmapId);
+            bool findPredicate(BeatmapInfo b) => b.OnlineBeatmapID == beatmapId;
+
+            PresentBeatmap(map.BeatmapSet, findPredicate);
+        }
+
+        public void SelectBeatmapSet(int setId)
```

Comment:
All the following methods you added seem to be used at a unique place, you may want to inline them instead as these are all fairly short